scanner/src/main/kotlin/experimental/NestedProvenanceScanResult.kt

```diff
@@ -136,4 +137,70 @@ data class NestedProvenanceScanResult(
 
         throw IllegalArgumentException("Could not find entry for $provenance.")
     }
+
+    /**
+     * Remove all scan results for [nestedProvenance]s which are not within the provided [path] and filter all findings
+     * within the scan results by [path], taking the sub repository paths into account.
+     * This also updates the VCS paths for all [RepositoryProvenance]s accordingly to make clear that they are not
+     * results for the whole repository.
+     *
+     * Throws an [IllegalArgumentException] if any provenance in [nestedProvenance] already has a VCS path set.
+     */
+    fun filterByVcsPath(path: String): NestedProvenanceScanResult {
+        if (path.isBlank()) return this
+
+        require(getProvenances().all { it !is RepositoryProvenance || it.vcsInfo.path.isBlank() }) {
+            "filterByVcsPath() requires that all provenances have no VCS path set."
```

Comment:
I wouldn't quote the function name here. Maybe better:

    "Can only filter repository / generic provenances by VCS path that were not created for a specific VCS path."